test/smartwallet/CustomSmartWallet.test.ts

```diff
@@ -0,0 +1,1197 @@
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        null,
+                        worker,
+                        '0',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize reverts with negative gas amount', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        token.address,
+                        worker,
+                        '100',
+                        '-400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize reverts with negative token amount', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isRejected(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        token.address,
+                        worker,
+                        '-2',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isFalse(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize successfully with token as 0x address parameter', async () => {
+                //Making sure the contract has not been initialized yet
+                assert.isFalse(await smartWallet.isInitialized());
+                //Initializing the contract
+                await assert.isFulfilled(
+                    smartWallet.initialize(
+                        senderAddress,
+                        customLogic.address,
+                        constants.ZERO_ADDRESS,
+                        worker,
+                        '10',
+                        '400000',
+                        '0x'
+                    )
+                );
+
+                assert.isTrue(await smartWallet.isInitialized());
+            });
+
+            it('Should verify method initialize successfully with worker address as null parameter', async () => {
```

Comment:
Is this a correct description for the test?

Reply:
Not valid anymore, because we fixed this in the [related PR](https://github.com/rsksmart/rif-relay-contracts/pull/33/) with a [specific commit](a410ef61f0de578ed2f9ac7b9dbd3a8a90992a05).